### PR TITLE
PyTorch implementation of VCT

### PR DESCRIPTION
**Pull request by desi-ivanova**
- add nn.Modules implementing analysis, synthesis and entropy model
    - create nn.Module pipeline to combine them
    - create pytorch_lightning model module and the data modules

### CODE REVIEW

**Comment by desi-ivanova** [comment]
@mmuckley done, and ran isort so checks pass :)

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -76,7 +76,6 @@ jobs:
           mypy --version
           mypy neuralcompression
           flake8 --version
-          flake8 neuralcompression tests projects
```
Comment (mmuckley) [inline .github/workflows/build-and-test.yml:79]: Sorry, could you add back this line? Just delete `projects`.

**Review by mmuckley — CHANGES_REQUESTED** [review]
Sorry, we do want to lint with `flake8`, we just don't need to do it on the `projects` folder haha.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (desi-ivanova) [reply .github/workflows/build-and-test.yml:79]: apologies for the misunderstanding

**Review by mmuckley — APPROVED** [review]
Overall LGTM. Code quality was previously reviewed internally. The primary validation is the rate-distortion performance point on UVG, which is presented in the README.

Thanks so much for working on this @desi-ivanova !

**Review by KarenUllrich — COMMENTED** [review]
LGTM

**Pull request merged**